contracts/tokenomics/generator/src/contract.rs

```diff
@@ -1638,6 +1653,37 @@ pub fn query_simulate_future_reward(
     Ok(simulated_reward)
 }
 
+/// ## Description
+/// Returns an [`ContractError`] on failure, otherwise returns a list of stakers for certain generator
+pub fn query_list_of_stakers(
+    deps: Deps,
+    lp_token: String,
+    start_after: Option<String>,
+    limit: Option<u32>,
+) -> Result<Vec<StakerResponse>, ContractError> {
+    let lp_addr = addr_validate_to_lower(deps.api, lp_token.as_str())?;
+    if POOL_INFO.has(deps.storage, &lp_addr) {
+        let res: AllAccountsResponse =
```

Comment:
Are you getting all addresses that have LP token and check if the lp token is present in generator? We would have a hard time finding the ones that do really use the generator and pagination doesn't make sense at all here. 
We need to iterate through USER_INFO by the first part of the key instead.